server/athenian/api/auth.py

```diff
@@ -117,16 +118,30 @@ def ensure_static_configuration(cls):
         await self._set_user(request)
         return await handler(request)
 
-    async def get_users(self, users: Iterable[str]) -> List[User]:
+    async def get_users(self, users: Sequence[str]) -> List[Optional[User]]:
         """Fetch several users by their IDs."""
         token = await self.mgmt_token()
-
-        async def get_user(user: str):
-            try:
-                resp = await self._session.get("https://%s/api/v2/users/%s" % (self._domain, user),
-                                               headers={"Authorization": "Bearer " + token})
-            except RuntimeError:
-                # our loop is closed and we are doomed
+        assert len(users) >= 0  # we need __len__
+
+        async def get_user(user: str) -> Optional[User]:
+            retries = 1
+            while retries <= 10:
```

Comment:
You could eventually use a `for-else` block to avoid handling counter and checking again the status:

```
for i in range(1, 11):
    try:
        ...
        if resp.status == HTTPStatus.TOO_MANY_REQUESTS:
            ...
        else:
            break
    except:
        return None
else:
    return None
```

Reply:
Much better this way!